packages/blob-storage/indexer/indexer.go

```diff
@@ -0,0 +1,388 @@
+}
+
+// InitFromConfig inits a new Indexer from a provided Config struct
+func InitFromConfig(ctx context.Context, i *Indexer, cfg *Config) (err error) {
+	client, err := ethclient.Dial(cfg.RPCURL)
+	if err != nil {
+		return err
+	}
+
+	taikoL1, err := taikol1.NewTaikoL1(cfg.ContractAddress, client)
+	if err != nil {
+		return err
+	}
+
+	db, err := mongodb.NewMongoDBClient(mongodb.MongoDBConfig{
+		Host:     cfg.DBHost,
+		Port:     cfg.DBPort,
+		Username: cfg.DBUsername,
+		Password: cfg.DBPassword,
+		Database: cfg.DBDatabase,
+	})
+	if err != nil {
+		return err
+	}
+
+	i.ethClient = client
+	i.beaconURL = cfg.BeaconURL
+	i.taikoL1 = taikoL1
+	i.startHeight = cfg.StartingBlockID
+	i.db = db
+	i.wg = &sync.WaitGroup{}
+	i.ctx = ctx
+	i.cfg = cfg
+
+	return nil
+}
+
+func (i *Indexer) Start() error {
+	i.wg.Add(1)
+
+	var processingBlockHeight uint64
+
+	if i.startHeight != nil {
+		processingBlockHeight = *i.startHeight
+	} else {
+		n, err := i.getLatestBlockID()
+		if err != nil {
+			return err
+		}
+
+		processingBlockHeight = n
+	}
+
+	// get the latest header
+	header, err := i.ethClient.HeaderByNumber(i.ctx, nil)
+	if err != nil {
+		return err
+	}
+
+	// the end block is the latest header.
+	endBlockID := header.Number.Uint64()
+
+	var defaultBlockBatchSize uint64 = 50
+
+	slog.Info("fetching batch block events",
+		"startHeight", processingBlockHeight,
+		"endblock", endBlockID,
+		"batchsize", defaultBlockBatchSize,
+	)
+
+	for j := processingBlockHeight; j < endBlockID; j += defaultBlockBatchSize {
+		end := processingBlockHeight + uint64(defaultBlockBatchSize)
+		// if the end of the batch is greater than the latest block number, set end
+		// to the latest block number
+		if end > endBlockID {
+			end = endBlockID
+		}
+
+		// filter exclusive of the end block.
+		// we use "end" as the next starting point of the batch, and
+		// process up to end - 1 for this batch.
+		filterEnd := end - 1
+
+		slog.Info("block batch", "start", j, "end", filterEnd)
+
+		opts := &bind.FilterOpts{
+			Start:   processingBlockHeight,
+			End:     &filterEnd,
+			Context: i.ctx,
+		}
+
+		group, groupCtx := errgroup.WithContext(i.ctx)
+
+		events, err := i.taikoL1.FilterBlockProposed(opts, nil, nil)
+		if err != nil {
+			return err
+		}
+
+		for events.Next() {
+			event := events.Event
+			group.Go(func() error {
+				if err := i.storeBlob(groupCtx, event); err != nil {
+					slog.Error("error storing blob", "error", err)
+				}
+
+				return nil
+			})
+		}
+
+		// wait for the last of the goroutines to finish
+		if err := group.Wait(); err != nil {
+			return err
+		}
+
+		processingBlockHeight += defaultBlockBatchSize
+	}
+	slog.Info(
+		"indexer fully caught up",
+	)
+
+	return i.subscribe(context.Background())
+}
+
+func (i *Indexer) subscribe(ctx context.Context) error {
+	defer func() {
+		i.wg.Done()
+	}()
+
+	slog.Info("subscribing to new events")
+
+	errChan := make(chan error)
+
+	go i.subscribeBlockProposed(ctx, errChan)
```

Comment:
Since the event subscription is not promising: https://github.com/taikoxyz/taiko-mono/pull/16353#discussion_r1519176189 , shall we keep using the new blocks scanning logic here?

Reply:
I didn't know that. We use subscriptions in the relayer and eventindexer as well. I should change it for both of them as well, then.  
  
I will do it for this PR first.

Reply:
I updated it to run on an event loop ticker instead, and crawl blocks. Please take a look. No more subscriptions.

I will do the same for our other services.